File: src/main/java/cn/chahuyun/economy/model/title/TitleTemplateSimpleImpl.java
package cn.chahuyun.economy.model.title;

import cn.chahuyun.economy.entity.TitleInfo;
import cn.chahuyun.economy.entity.UserInfo;
import lombok.Getter;

/**
 * 简单称号模板实现
 */
@Getter
public class TitleTemplateSimpleImpl extends TitleTemplate {

    private final Boolean gradient;
    private final Boolean impactName;
    private final String title;
    private final String sColor;
    private final String eColor;

    public TitleTemplateSimpleImpl(String templateCode, Integer validityPeriod, String titleName, Double price, Boolean gradient, Boolean impactName, String title, String sColor, String eColor) {
        super(templateCode, titleName, validityPeriod, true, price);
        this.gradient = gradient;
        this.impactName = impactName;
        this.title = title;
        this.sColor = sColor;
        this.eColor = eColor;
    }

    public TitleTemplateSimpleImpl(String templateCode, Integer validityPeriod, String titleName, Boolean canIBuy, Double price, Boolean gradient, Boolean impactName, String title, String sColor, String eColor) {
        super(templateCode, titleName, validityPeriod, canIBuy, price);
        this.gradient = gradient;
        this.impactName = impactName;
        this.title = title;
        this.sColor = sColor;
        this.eColor = eColor;
    }

    @Override
    public TitleInfo createTitleInfo(UserInfo userInfo) {
        return TitleInfo.builder()
                .userId(userInfo.getQq())
                .code(getTemplateCode())
                .name(getTitleName())
                .status(true)
                .impactName(impactName)
                .gradient(gradient)
                .title(title)
                .sColor(sColor)
                .eColor(eColor)
                .build();
    }
}

File: src/main/kotlin/cn/chahuyun/economy/model/title/TitleTemplateSimpleImpl.kt
package cn.chahuyun.economy.model.title

import cn.chahuyun.economy.entity.TitleInfo
import cn.chahuyun.economy.entity.UserInfo

/**
 * 简单称号模板实现
 */
class TitleTemplateSimpleImpl(
    templateCode: String,
    validityPeriod: Int?,
    titleName: String,
    canIBuy: Boolean = true,
    price: Double?,
    val gradient: Boolean? = null,
    val impactName: Boolean? = null,
    val title: String = "",
    val sColor: String = "",
    val eColor: String = ""
) : TitleTemplate(templateCode, titleName, validityPeriod, canIBuy, price) {

    constructor(
        templateCode: String,
        validityPeriod: Int?,
        titleName: String,
        price: Double?,
        gradient: Boolean?,
        impactName: Boolean?,
        title: String,
        sColor: String,
        eColor: String
    ) : this(
        templateCode = templateCode,
        validityPeriod = validityPeriod,
        titleName = titleName,
        canIBuy = true,
        price = price,
        gradient = gradient,
        impactName = impactName,
        title = title,
        sColor = sColor,
        eColor = eColor
    )

    override fun createTitleInfo(userInfo: UserInfo): TitleInfo {
        return TitleInfo(userInfo.qq, templateCode, titleName, title).apply {
            status = true
            this.impactName = this@TitleTemplateSimpleImpl.impactName ?: false
            this.gradient = this@TitleTemplateSimpleImpl.gradient ?: false
            this.sColor = this@TitleTemplateSimpleImpl.sColor
            this.eColor = this@TitleTemplateSimpleImpl.eColor
        }
    }
}
